### PR TITLE
Fire "dataloading" events from all resources that fire "data" events

### PR DESCRIPTION
**Pull request by lucaswoj**
This PR completes our support for "dataloading" events, ensuring they are fired from all resources that fire "data" events.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] document any changes to public APIs
 - [x] manually test the debug page

### CODE REVIEW

**Comment by lucaswoj** [comment]
cc @jfirebaugh @mollymerp @lbud for review

File: js/util/evented.js [hunk 1]
```diff
@@ -95,6 +96,18 @@ var Evented = {
...
     /**
+     * Asyncronously fires an event of the specified type. The event will be
+     * fired within a "setTimeout(..., 0)" callback.
+     *
+     * @param {string} type The type of event to fire.
+     * @param {Object} [data] Data to be passed to any listeners.
+     * @returns {Object} `this`
+     */
+    asyncFire: function(type, data) {
+        window.setTimeout(this.fire.bind(this, type, data), 0);
```
Comment (lucaswoj) [inline js/util/evented.js]: @jfirebaugh How do you feel about this use of `setTimeout`? There are some situations where we cannot fire an event synchronously (namely in a constructor).

Comment (jfirebaugh) [inline js/util/evented.js]: Well, it's probably technically a race condition -- I don't think the DOM specs guarantee that the `setTimeout(0)` will trigger before the AJAX request onload (which is plausible if, say, the request is immediately satisfied by a cache).

Comment (jfirebaugh) [inline js/util/evented.js]: Passing the evented parent as a parameter to each of these constructors, calling `setEventedParent`, then firing a sync `dataloading`?

Comment (lucaswoj) [inline js/util/evented.js]: I like that design 👍 I'll explore first thing tomorrow.

Comment (lucaswoj) [inline js/util/evented.js]: Done. This turned out to be quite clean.

**Comment by lucaswoj** [comment]
Ready for another round of 👁

**Pull request merged**